Cookbook/Technical-Documents/XcodeTips.md

```diff
@@ -1,20 +1,70 @@
+- **Cmd-Ctrl-J** (as in “jump”) or **Cmd-Ctrl-Click** on a symbol to go to definition. Add **Opt** to open in assistant editor.
+- Right click and “Find Selected Symbol in Workspace” to show in the search pane where the symbol under cursor is used.
+- **Cmd-Shift-F** to search in workspace (the whole project). You'll want to select Text on top because searching for references is tricky manually, they require specific syntax.
+- **Cmd-Opt-J** and type to filter files by name in the Project Navigator (or tests in the Test Navigator). Useful if you want to find multiple files with the same name, or can't quite remember how a file is called.
+
+##### File level navigation
+- **Cmd-Shift-J** to show the open file in the Project Navigator.
+- **Cmd-Ctrl-Left** or **Right** in a file to navigate between recently open files.
+- **Ctrl-6** in a file to open the list of methods and variables. You can then type to apply a filter and hit Enter to jump to the selected completion.
+- **Cmd-Opt-Shift-Left** in a file to fold all methods and **Cmd-Opt-Shift-Right** to unfold. May be useful to get a quick overview.
+
+#### Making space for your code
+Use these keybindings to give your code more space, and show the panes again quickly when needed. You can keep the panes nice and wide this way, and just hide them when you're not using them, rather than keep them thin and see them all the time.
+
+- **Cmd-Shift-Y** to show or hide the bottom drawer (locals and console output).
+- **Cmd-0** and **Cmd-Opt-0** to fold and unfold the left and right panes, respectively.
+- **Cmd-Enter** to close the assistant editor and leave only the main editor pane.
+
+#### Working with the assistant editor
+If your screen is big enough or your font is small enough to allow using two editors side by side, this is a very efficient way to work.
+
+- **Cmd-Enter** to close the assistant editor and leave only the main editor pane.
+- **Ctrl-`** to switch between the main editor and the assistant editor.
+
+In general, you can add **Opt** to any command that opens or navigates to a file to open it in the assistant editor instead. Here are a few to get you started:
+
+- **Cmd-Ctrl-Opt-J** or **Cmd-Ctrl-Opt-Click** to open the definition for symbol under cursor in the assistant editor.
+- **Opt-Click** in workspace search (**Cmd-Shift-F**).
+- **Opt-Enter** in Open Quickly (**Cmd-Shift-O**).
+
+## Testing
+Everyone knows you hit **Cmd-U** to start testing… right?
+
+It's annoying when your tests fail, but even more annoying to try and compare the failed snapshots in */tmp/SnapshotFailures/*. Thankfully, Xcode can help you out.
+
+After you've run the tests, in the Test Navigator (**Cmd-9**) open the latest test report, switch to Failed tests and you'll find XCTAttachments for before, after and diff attached to the failed test:
+
+![Test Navigator Attachments](./Assets/test-navigator-attachments.png)
```

Comment:
```suggestion
![Test Navigator Attachments](./Assets/test-navigator-attachments.png)

You can view the same report on the CircleCI by navigating to the job's Artifacts and opening `Users/distiller/project/output/scan/index.html` [file](https://147627-18095916-gh.circle-artifacts.com/0/Users/distiller/project/output/scan/index.html)
```

Reply:
Hmm, I don't think it makes much sense to actually _link_ to a CI result as I expect it will be deleted eventually. I'd rather attach another screenshot.

But it's a good place to make a note about CI, I agree with that.

Reply:
Added in https://github.com/Babylonpartners/ios-playbook/pull/205/commits/168646e75ef7ff023ce534a7acd2e8ae1700ba84